pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -446,13 +515,37 @@ func (r *reconciler) computeOFRulesForAdd(rule *CompletedRule, ofPriority *uint1
 	r.lastRealizeds.Store(rule.ID, lastRealized)
 
 	ofRuleByServicesMap := map[servicesKey]*types.PolicyRule{}
-
+	groupAddresses := r.getIGMPGroupAddress(rule)
+	isIGMP := len(groupAddresses) > 0
+	var from1, from2 []types.Address
+	if isIGMP && rule.Direction == v1beta2.DirectionIn {
+		// IGMP query
+		svcKey := servicesKey(fmt.Sprintf("igmp-query-%s", rule.ID))
```

Comment:
servicesKey is used to resolve named port case, which doesn't apply to IGMP and multicast, you could just use a dummy key here, maybe create a constant `igmpServicesKey = "igmp-services-key"` and use it for all rules (the key is just used in the scope of the rule itself)